File: go/0274_hindex.go
func hIndex(citations []int) int {
	buckets := make([]int, len(citations)+1)
	for _, c := range citations {
		if c >= len(citations) {
			buckets[len(buckets)-1]++
			continue
		}
		buckets[c]++
	}

	sofar := 0
	for i := len(buckets) - 1; i >= 0; i-- {
		sofar += buckets[i]
		if sofar >= i {
			return i
		}
	}
	return 0
}

File: python/0274_h_index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        clen = len(citations)
        if clen == 0:
            return 0

        citations.sort(reverse=True)
        i = 0
        while i < clen:
            if i + 1 <= citations[i]:
                i += 1
                continue
            break
        return i


class Solution:
    def hIndex(self, citations: List[int]) -> int:
        clen = len(citations)
        if clen == 0:
            return 0

        buckets = [0] * (clen + 1)
        for c in citations:
            buckets[min(c, clen)] += 1

        i = len(buckets) - 1
        s = 0
        while i >= 0:
            s += buckets[i]
            if s >= i:
                break
            i -= 1
        return i
